plugins/flatpak/gs-flatpak.c

```diff
@@ -2477,18 +2477,82 @@ gs_flatpak_app_remove_source (GsFlatpak *self,
 	return TRUE;
 }
 
+/*
+ * gs_flatpak_get_services_app_for_runtime:
+ * Specific to Endless runtimes, gets a GsApp reference for the related
+ * "EknServices" app that provides system services for apps that use the
+ * given runtime.
+ *
+ * Returns NULL if no services app was required or there was an error.
+ * Does not take a GError since any error should be recoverable. However, in
+ * order to make sure the operation was not cancelled, you should check the
+ * status of the GCancellable after calling this.
+ */
+static GsApp *
+gs_flatpak_get_services_app_for_runtime (GsFlatpak *self, GsApp *runtime,
+                                         GCancellable *cancellable)
+{
+	const gchar *runtime_id, *runtime_branch, *services_branch, *arch;
```

Comment:
Usually we don't have declarations like this in g-s. Even if I prefer these actually.
I am also not sure if the C99 standard will complain or not, but I'd make these several lines. (Just a nitpick though, not a blocker for me, unless there's a compiler warning)

Reply:
Multiple declarations per line have been a thing in C since at least [1978](https://archive.org/details/TheCProgrammingLanguageFirstEdition) :smile: But I agree it's better to keep the style of the surrounding code.